packages/search-adapter/src/empathy/__tests__/response-mappers/empathy-facet-mapper.spec.ts

```diff
@@ -107,21 +117,25 @@ function expectFacetToMatchMock(mappedFacet: HierarchicalFacet | SimpleFacet | N
 function expectBooleanFiltersToHaveSelectedPropertyTo(filters: BooleanFilter[], selected: boolean | null): void {
   filters.forEach(filter => {
     expect(filter.selected).toEqual(selected);
-    if (isHierarchicalFilter(filter) && !!filter.children) {
-      expectBooleanFiltersToHaveSelectedPropertyTo(filter.children, selected);
-    }
   });
 }
 
 function expectAllFiltersToBeValid(filters: Filter[], schema: HierarchicalFilter | SimpleFilter | NumberRangeFilter): void {
   return filters.forEach(filter => {
     expect(filter).toMatchObject(schema);
-    if (isHierarchicalFilter(filter) && filter.children) {
-      expectAllFiltersToBeValid(filter.children, HierarchicalFilterSchema);
-    }
   });
 }
 
-function isHierarchicalFilter(filter: Filter): filter is HierarchicalFilter {
-  return filter.modelName === 'HierarchicalFilter';
+function countHierarchicalChildren(filter: EmpathyFacet | EmpathyFilter): number {
+  let counter = 0;
+  if (filter.values?.length) {
+    for (let value of filter.values) {
+      counter += countHierarchicalChildren(value);
+    }
+  }
+  return counter + (filter.values?.length || 0);
+}
+
+function findFilterInChildren({ parentId, id }: HierarchicalFilter, allFilters: HierarchicalFilter[]) {
+  return parentId !== null ? allFilters.some(filter => filter.children?.some(child => child === id)) : false;
```

Comment:
Same happens for this, I don't get this function, you are using it to search that a filter that is in an array is contained in that array?  Below line doesn't make any sense.

```
findFilterInChildren(mappedFacet.filters[1], mappedFacet.filters)
```

Reply:
No, I am using this to find if a filter of the array is contained in any of the children arrays of any of the filters. I think this is a valid validation xD But I'll try to simplify this.